### PR TITLE
Edits in documentation

### PR DESCRIPTION
**Pull request by migueldvb**
### Description

This pull request includes some updates in the development documentation and a few minor corrections.

### CODE REVIEW

**Review by pllim — APPROVED** [review]
Wow, good catch. Thanks!

**Comment by pllim** [comment]
This warning might be related and should be fixed before merge:
```
.../docs/index.rst:17: WARNING: Duplicate explicit target name: "affiliated packages".
```

**Comment by migueldvb** [comment]
> `.../docs/index.rst:17: WARNING: Duplicate explicit target name: "affiliated packages".`

Ah! it was introduced in this PR. I think it should be fixed now.

File: docs/index.rst [hunk 1]
```diff
@@ -176,3 +176,4 @@ Index
 * :ref:`search`
 
 .. _feedback@astropy.org: mailto:feedback@astropy.org
+.. _affiliated packages: http://www.astropy.org/affiliated/
```
Comment (pllim) [inline docs/index.rst:179]: You HTTPS-ed everything else, why not here too? 😉 
```suggestion
.. _affiliated packages: https://www.astropy.org/affiliated/
```

Comment (migueldvb) [reply docs/index.rst]: That's right! I amended the last commit.

**Review by pllim — APPROVED** [review]
Perfect. Thanks!

**Pull request merged**